diff -ruN org/zapret/nfq/desync.c patch/zapret/nfq/desync.c
--- org/zapret/nfq/desync.c	2024-11-24 17:52:38.389942520 -0600
+++ patch/zapret/nfq/desync.c	2024-11-24 17:53:49.119550832 -0600
@@ -2251,6 +2251,18 @@
 	if (!!dis.ip != !!dis.ip6)
 	{
 		packet_debug(replay, &dis);
+		if (params.force_ttl)
+		{
+			if (dis.ip)
+			{
+				dis.ip->ip_ttl = params.ttl;
+				ip4_fix_checksum(dis.ip);
+			}
+			else
+			{
+				dis.ip6->ip6_ctlun.ip6_un1.ip6_un1_hlim = params.ttl;
+			}
+		}
 		switch(dis.proto)
 		{
 			case IPPROTO_TCP:
@@ -2269,6 +2281,10 @@
 				break;
 		}
 		*len_pkt = dis.len_pkt;
+		if (params.force_ttl && (verdict & VERDICT_MASK) == VERDICT_PASS)
+		{
+			verdict = VERDICT_MODIFY;
+		}
 	}
 	return verdict;
 }
diff -ruN org/zapret/nfq/nfqws.c patch/zapret/nfq/nfqws.c
--- org/zapret/nfq/nfqws.c	2024-11-24 17:52:38.389942520 -0600
+++ patch/zapret/nfq/nfqws.c	2024-11-24 17:57:57.388161032 -0600
@@ -1003,6 +1003,8 @@
 #ifdef __linux__
 		" --bind-fix4\t\t\t\t\t; apply outgoing interface selection fix for generated ipv4 packets\n"
 		" --bind-fix6\t\t\t\t\t; apply outgoing interface selection fix for generated ipv6 packets\n"
+		" --force-ttl\t\t\t\t\t; overwrite the ttl/hl for traffic passing through nfqws\n"
+		" --ttl=<int>\t\t\t\t\t; ttl/hl applied by --force-ttl, default is 64\n"
 #endif
 		" --ctrack-timeouts=S:E:F[:U]\t\t\t; internal conntrack timeouts for TCP SYN, ESTABLISHED, FIN stages, UDP timeout. default %u:%u:%u:%u\n"
 #ifdef __CYGWIN__
@@ -1179,6 +1181,7 @@
 	params.ctrack_t_est = CTRACK_T_EST;
 	params.ctrack_t_fin = CTRACK_T_FIN;
 	params.ctrack_t_udp = CTRACK_T_UDP;
+	params.ttl = 64;
 	
 	LIST_INIT(&params.hostlists);
 	LIST_INIT(&params.ipsets);
@@ -1285,6 +1288,8 @@
 #ifdef __linux__
 		{"bind-fix4",no_argument,0,0},		// optidx=61
 		{"bind-fix6",no_argument,0,0},		// optidx=62
+		{"force-ttl",no_argument,0,0},		// optidx=63
+		{"ttl",required_argument,0,0},		// optidx=64
 #elif defined(__CYGWIN__)
 		{"wf-iface",required_argument,0,0},	// optidx=61
 		{"wf-l3",required_argument,0,0},	// optidx=62
@@ -1894,6 +1899,17 @@
 		case 62: /* bind-fix6 */
 			params.bind_fix6 = true;
 			break;
+		case 63: /* force_ttl */
+			params.force_ttl = true;
+			break;
+		case 64: /* ttl */
+			params.ttl = (uint8_t)atoi(optarg);
+			if (params.ttl < 1 || params.ttl > 255)
+			{
+				fprintf(stderr, "packet ttl is not valid\n");
+				exit_clean(1);
+			}
+			break;
 #elif defined(__CYGWIN__)
 		case 61: /* wf-iface */
 			if (!sscanf(optarg,"%u.%u",&IfIdx,&SubIfIdx))
diff -ruN org/zapret/nfq/params.h patch/zapret/nfq/params.h
--- org/zapret/nfq/params.h	2024-11-24 17:52:38.389942520 -0600
+++ patch/zapret/nfq/params.h	2024-11-24 17:58:30.191309348 -0600
@@ -142,6 +142,9 @@
 
 	unsigned int ctrack_t_syn, ctrack_t_est, ctrack_t_fin, ctrack_t_udp;
 	t_conntrack conntrack;
+
+	char force_ttl;
+	uint8_t ttl;
 };
 
 extern struct params_s params;
