diff -ruN org/jni/Android.mk patch/jni/Android.mk
--- org/jni/Android.mk	2024-03-06 22:47:28.148607162 -0600
+++ patch/jni/Android.mk	2024-03-06 22:50:36.561934193 -0600
@@ -39,4 +39,4 @@
 LOCAL_CFLAGS += -mfpu=neon
 endif
 LOCAL_STATIC_LIBRARIES := yaml hev-task-system
-include $(BUILD_SHARED_LIBRARY)
+include $(BUILD_EXECUTABLE)
diff -ruN org/jni/Application.mk patch/jni/Application.mk
--- org/jni/Application.mk	2024-03-06 23:03:52.741907698 -0600
+++ patch/jni/Application.mk	2024-03-06 23:04:48.011905863 -0600
@@ -14,7 +14,7 @@
 #
 
 APP_OPTIM := release
-APP_PLATFORM := android-21
+APP_PLATFORM := android-24
 APP_ABI := armeabi-v7a arm64-v8a
 APP_CFLAGS := -O3
 NDK_TOOLCHAIN_VERSION := clang
