diff -ruN org/dnsmasq-2.90/Makefile patch/dnsmasq-2.90/Makefile
--- org/dnsmasq-2.90/Makefile	2024-02-13 07:49:15.000000000 -0600
+++ patch/dnsmasq-2.90/Makefile	2024-03-08 02:15:32.309081848 -0600
@@ -24,8 +24,8 @@
 LOCALEDIR     = $(PREFIX)/share/locale
 BUILDDIR      = $(SRC)
 DESTDIR       = 
-CFLAGS        = -Wall -W -O2
-LDFLAGS       = 
+CFLAGS        = -Wall -W -O2 -fPIC
+LDFLAGS       = -llog
 COPTS         = 
 RPM_OPT_FLAGS = 
 LIBS          = 
diff -ruN org/dnsmasq-2.90/src/dhcp.c patch/dnsmasq-2.90/src/dhcp.c
--- org/dnsmasq-2.90/src/dhcp.c	2024-02-13 07:49:15.000000000 -0600
+++ patch/dnsmasq-2.90/src/dhcp.c	2024-03-08 02:09:53.155759631 -0600
@@ -16,6 +16,10 @@
 
 #include "dnsmasq.h"
 
+#ifdef __ANDROID__
+#define ETHER_ADDR_LEN 6
+#endif
+
 #ifdef HAVE_DHCP
 
 struct iface_param {
diff -ruN org/dnsmasq-2.90/src/tftp.c patch/dnsmasq-2.90/src/tftp.c
--- org/dnsmasq-2.90/src/tftp.c	2024-02-13 07:49:15.000000000 -0600
+++ patch/dnsmasq-2.90/src/tftp.c	2024-03-08 02:10:31.275758376 -0600
@@ -16,6 +16,8 @@
 
 #include "dnsmasq.h"
 
+#define ETHER_ADDR_LEN 6
+
 #ifdef HAVE_TFTP
 
 static void handle_tftp(time_t now, struct tftp_transfer *transfer, ssize_t len);
