diff -ruN org/dnsmasq-2.91/Makefile patch/dnsmasq-2.91/Makefile
--- org/dnsmasq-2.91/Makefile	2025-03-14 10:09:35.000000000 -0500
+++ patch/dnsmasq-2.91/Makefile	2025-03-31 23:52:40.927904166 -0500
@@ -24,8 +24,8 @@
 LOCALEDIR     = $(PREFIX)/share/locale
 BUILDDIR      = $(SRC)
 DESTDIR       = 
-CFLAGS        = -Wall -W -O2
-LDFLAGS       = 
+CFLAGS        = -Wall -W -O2 -fPIC
+LDFLAGS       = -llog
 COPTS         = 
 RPM_OPT_FLAGS = 
 LIBS          = 
diff -ruN org/dnsmasq-2.91/src/dhcp.c patch/dnsmasq-2.91/src/dhcp.c
--- org/dnsmasq-2.91/src/dhcp.c	2025-03-14 10:09:35.000000000 -0500
+++ patch/dnsmasq-2.91/src/dhcp.c	2025-03-31 23:53:49.272491755 -0500
@@ -16,6 +16,10 @@
 
 #include "dnsmasq.h"
 
+#ifdef __ANDROID__
+#define ETHER_ADDR_LEN 6
+#endif
+
 #ifdef HAVE_DHCP
 
 struct iface_param {
diff -ruN org/dnsmasq-2.91/src/tftp.c patch/dnsmasq-2.91/src/tftp.c
--- org/dnsmasq-2.91/src/tftp.c	2025-03-14 10:09:35.000000000 -0500
+++ patch/dnsmasq-2.91/src/tftp.c	2025-03-31 23:54:36.758208856 -0500
@@ -16,6 +16,10 @@
 
 #include "dnsmasq.h"
 
+#ifdef __ANDROID__
+#define ETHER_ADDR_LEN 6
+#endif
+
 #ifdef HAVE_TFTP
 
 static void handle_tftp(time_t now, struct tftp_transfer *transfer, ssize_t len);
